pkg/cmd/addon/index.go

```diff
@@ -262,12 +262,14 @@ func addDefaultIndex() error {
 	}
 
 	defaultIndexDir := path.Join(addonDir, types.KubeBlocksReleaseName)
-	if _, err := os.Stat(defaultIndexDir); err != nil && os.IsNotExist(err) {
+	if _, err := os.Stat(defaultIndexDir); os.IsNotExist(err) {
 		if err = util.EnsureCloned(types.DefaultAddonIndexURL, defaultIndexDir); err != nil {
 			return err
 		}
 		fmt.Printf("Default addon index \"kubeblocks\" has been added.")
 		return nil
+	} else if err != nil {
+		return err
 	}
 	return fmt.Errorf("default index %s:%s already exists", types.KubeBlocksReleaseName, types.DefaultAddonIndexURL)
```

Comment:
I do not think we should throw an error if default index already exits. Better to check if the default index already exists first like plugin do, if not, add the default index.

Reply:
ok, I forget that the default addon repo will be added when using index cmd including it's subcommand.